eslint-plugin-expensify/onyx-props-must-have-default.js

```diff
@@ -0,0 +1,248 @@
+                        resolvedProps.push(...externalProps);
+                    } else {
+                        // If the spread operator variable is in same file
+                        const internalProps = variableNode.defs[0].node.init.properties;
+                        resolvedProps.push(...internalProps);
+                    }
+                },
+            );
+            return resolvedProps;
+        }
+
+        // Get the final list of properties
+        function getPropsValue(scope, propsVar) {
+            const variables = scope.variables;
+            let props = getVariableNode(variables, propsVar);
+            if (!props) {
+                // Required in case of HOCs like withPolicy, since they might have the prop types
+                // declared outside the function, which is not in function scope
+                props = getVariableNode(scope.upper.variables, propsVar);
+            }
+
+            if (!propsVar || !props) {
+                return undefined;
+            }
+
+            // Need to check for existence of init since it is not present if the assignment is to an imported variable
+            let propsProperties;
+            if (props.defs[0].node.type === 'VariableDeclarator') {
+                // If the variable is an assignment
+                if (props.defs[0].node.init.type === 'ObjectExpression') {
+                    // let defaultProps = {....}
+                    propsProperties = props.defs[0].node.init && props.defs[0].node.init.properties;
+                } else if (props.defs[0].node.init.type === 'Identifier') {
+                    // let defaultProps = importedDefaultProps;
+                    return getPropsValue(scope, props.defs[0].node.init.name);
+                }
+
+                if (props.defs[0].type === 'ImportBinding') {
+                    propsProperties = getExternalProperties(props.defs[0].node.parent.source.value, props, props.defs[0].node.type);
+                } else if (hasSpreadOperator(props.defs[0].node)) {
+                    const resolvedProps = getProperties(scope, propsProperties);
+                    propsProperties = propsProperties.concat(resolvedProps);
+                }
+            } else if (props.defs[0].node.type === 'ImportSpecifier') {
+                // If the variable is a direct import
+                propsProperties = getExternalProperties(props.defs[0].node.parent.source.value, props, props.defs[0].node.type);
+            }
+
+            return propsProperties;
+        }
+
+        // Report and exit if direct assignment
+        // For example, Component.propType = {prop1: type1}
+        function isDirectAssignment(node, declaration, key) {
+            if (lodashGet(declaration, 'expression.right.type') === 'ObjectExpression') {
+                makeReport(node, MUST_USE_VARIABLE_FOR_ASSIGNMENT, key);
+                return 1;
+            }
+            return 0;
```

Comment:
Is there any reason to return this _1_ or _0_? Is possible we just return a boolean?